Client/include/RPCClient.h

```diff
@@ -0,0 +1,129 @@
+//
+// Created by kfeng on 12/26/22.
+//
+
+#ifndef CHRONOLOG_RPCCLIENT_H
+#define CHRONOLOG_RPCCLIENT_H
+
+#include <iostream>
+#include <unordered_map>
+#include <functional>
+#include <thallium.hpp>
+#include <sys/types.h>
+#include <unistd.h>
+#include "macro.h"
+#include "RPCFactory.h"
+#include "global_var_client.h"
+
+class RPCClient {
+public:
+    RPCClient() {
+        LOGD("%s constructor is called", typeid(*this).name());
+        rpc = ChronoLog::Singleton<ChronoLogRPCFactory>::GetInstance()
+                ->GetRPC(CHRONOLOG_CONF->RPC_BASE_SERVER_PORT);
+        g_RPC = rpc;
+        set_prefix("ChronoLog");
+        LOGD("%s constructor finishes, object created@%p in thread PID=%d",
+             typeid(*this).name(), this, getpid());
+    }
+
```

Comment:
client.h and RPCClient.h have a lot of common functions. You may want to consider combining them into a single class.

Reply:
Client is the public client API for ChronoLog and will contain more than just RPC Client for communicating with the Visor.  having RPCClient as a member of Client would give us more flexibility to tweak implementation without affecting the ClientAPI. I would not combine them.
Though I suggest that RPCClient does not include header files from the ChronoAPI directory. any header files that are included here will have to be exposed to the client . I suggest that anything RPCClient.h needs is defined directly in this file or eliminated.